src/main/java/com/mate/fruitshop/strategy/ProcessReturnTransaction.java

```diff
@@ -0,0 +1,25 @@
+package com.mate.fruitshop.strategy;
+
+import com.mate.fruitshop.dao.FruitStorageDao;
+import com.mate.fruitshop.dao.impl.FruitStorageDaoImpl;
+import com.mate.fruitshop.model.FruitEntry;
+import com.mate.fruitshop.model.Transaction;
+
+public class ProcessReturnTransaction implements TransactionProcessingStrategy {
+    private final FruitStorageDao dao;
+
+    public ProcessReturnTransaction() {
+        this.dao = new FruitStorageDaoImpl();
+    }
+
+    @Override
+    public boolean process(Transaction transaction) {
+        FruitEntry fruitEntry = dao.getEntryByFruitName(transaction.getFruitName());
+        if (fruitEntry == null) {
+            return dao.addFruitEntry(new FruitEntry(transaction.getFruitName(),
+                    transaction.getQuantity()));

```

Comment:
Again, how do we return fruits, that even were not on our balance?
What if customer returned us potatoes instead of oranges? ;)

Reply:
Customer can be returning potatoes that he bought 10 days ago, and they are already not on balance.
They would be rotten at this point, though)